terasoluna-gfw-common-libraries/terasoluna-gfw-common/src/test/java/org/terasoluna/gfw/common/codelist/i18n/SimpleI18nCodeListTest.java

```diff
@@ -116,51 +101,86 @@ public void testSetRowsByCodeList01() {
     }
 
     @Test(expected = UnsupportedOperationException.class)
-    public void testSetRowsByCodeList02() {
-        // check unmodifiable
-        testSetRowsByCodeList01.asMap(Locale.ENGLISH).put("0", "Sunday");
+    public void testAsMapCheckUnmodifiable() {
+        testSetRows.asMap(Locale.ENGLISH).put("0", "Sunday");
     }
 
     @Test
-    public void testSetColumns01() {
-        assertThat(testSetColumns01.codeListTable.size(), is(14)); // 2 rows x 7
-                                                                   // columns
+    public void testSetRows() {
+        assertThat(testSetRows.codeListTable.size(), is(14)); // 2 rows x 7
+                                                              // columns
+    }
 
-        Map<String, String> row1 = testSetColumns01.asMap(Locale.ENGLISH);
-        assertThat(row1, is(notNullValue()));
-        assertThat(row1.keySet().toString(), is("[0, 1, 2, 3, 4, 5, 6]")); // check
-                                                                           // order
-        assertThat(row1.get("0"), is("Sun."));
-        assertThat(row1.get("1"), is("Mon."));
-        assertThat(row1.get("2"), is("Tue."));
-        assertThat(row1.get("3"), is("Wed."));
-        assertThat(row1.get("4"), is("Thu."));
-        assertThat(row1.get("5"), is("Fri."));
-        assertThat(row1.get("6"), is("Sat."));
+    @Test
+    public void testSetRowsByCodeList() {
+        assertThat(testSetRowsByCodeList.codeListTable.size(), is(14)); // 2 rows x 7
+                                                                        // columns
+    }
 
-        Map<String, String> row2 = testSetColumns01.asMap(Locale.JAPANESE);
-        assertThat(row2, is(notNullValue()));
-        assertThat(row2.keySet().toString(), is("[0, 1, 2, 3, 4, 5, 6]")); // check
-                                                                           // order
-        assertThat(row2.get("0"), is("日"));
-        assertThat(row2.get("1"), is("月"));
-        assertThat(row2.get("2"), is("火"));
-        assertThat(row2.get("3"), is("水"));
-        assertThat(row2.get("4"), is("木"));
-        assertThat(row2.get("5"), is("金"));
-        assertThat(row2.get("6"), is("土"));
+    @Test
+    public void testSetColumns() {
+        assertThat(testSetColumns.codeListTable.size(), is(14)); // 2 rows x 7
+                                                                 // columns
     }
 
-    @Test(expected = UnsupportedOperationException.class)
-    public void testSetColumnss02() {
-        // check unmodifiable
-        testSetColumns01.asMap(Locale.ENGLISH).put("0", "Sunday");
+    @Test
+    public void testSetFallbackTo() {
+        testSetFallbackTo.setFallbackTo(Locale.JAPANESE);
+        testSetFallbackTo.afterPropertiesSet();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testSetFallbackToNull() {
+        testSetFallbackTo.setFallbackTo(null);
+        testSetFallbackTo.afterPropertiesSet();
```

Comment:
- [ ] Test cases should be implemented independently. You should not change state of bean `testSetFallbackTo`.

There are 2 ways.

* Create bean instance in each test case.
```java
    @Test
    public void testSetFallbackToNull2() {
        SimpleI18nCodeList codeList = new SimpleI18nCodeList();
        try {
            codeList.setFallbackTo(null);
        } catch (IllegalArgumentException e) {
            assertThat(e.getMessage(), is("fallbackTo must not be null"));
        }
    }
```

* Define `lazy-init` bean and get using `ApplicationObjectSupport`.

```xml
  <bean id="CL_testSetFallbackToNull" class="org.terasoluna.gfw.common.codelist.i18n.SimpleI18nCodeList"
        lazy-init="true">
    ....
    <property name="fallbackTo">
        <null />
    </property>
  </bean>
```

```java
public class SimpleI18nCodeListTest extends ApplicationObjectSupport {
    ....
    @Test
    public void testSetFallbackToNull() {
        try {
            super.getApplicationContext().getBean("CL_testSetFallbackToNull");
            fail("BeanCreationException not occered.");
        } catch (BeanCreationException e) {
            Throwable cause = e.getCause();
            assertThat(cause, instanceOf(PropertyBatchUpdateException.class));
            PropertyAccessException[] propertyAccessExceptions = ((PropertyBatchUpdateException) cause).getPropertyAccessExceptions();
            assertThat(propertyAccessExceptions, notNullValue());
            assertThat(propertyAccessExceptions.length, is(1));
            Throwable rootCause = propertyAccessExceptions[0].getCause();
            assertThat(rootCause, instanceOf(IllegalArgumentException.class));
            assertThat(rootCause.getMessage(), is("fallbackTo must not be null"));
        }
    }
```

And if you need to assert exception other than type, you may consider introducing `ExpectedException` rule.

Reply:
Modified by f22d57837d95ab1e8ba00dfbdb65a1814247699d

`testSetFallbackToNull` has adopted Case 1.
In `testSetFallbackToInvalidLanguage` and `testSetFallbackToInvalidLanguageMatchingNation`, Case 2 was adopted because it was difficult to implement in Case 1.